<?xml version='1.0' encoding='UTF-8'?>
<root version="2.3.8">
 <tabbed_widget name="Main Window" parent="main_window">
  <plotmatrix tab_name="plot" rows="2" columns="2">
   <plot row="0" style="Lines" col="0">
    <range bottom="-0.314429" left="43.410000" top="-0.314429" right="48.410000"/>
    <limitY/>
    <curve name="/wheel_leg/joint_states/joint_base_right_leg1/position" custom_transform="noTransform" B="180" G="119" R="31"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="1" style="Lines" col="0">
    <range bottom="0.199671" left="43.410000" top="0.199759" right="48.410000"/>
    <limitY/>
    <curve name="/wheel_leg/joint_states/joint_base_right_leg1/effort" custom_transform="noTransform" B="40" G="39" R="214"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="0" style="Lines" col="1">
    <range bottom="1.047192" left="43.410000" top="1.047192" right="48.410000"/>
    <limitY/>
    <curve name="/wheel_leg/joint_states/joint_right_leg1_leg2/position" custom_transform="noTransform" B="56" G="201" R="26"/>
    <transform value="noTransform"/>
   </plot>
   <plot row="1" style="Lines" col="1">
    <range bottom="-0.100000" left="43.410000" top="0.100000" right="48.410000"/>
    <limitY/>
    <curve name="/wheel_leg/joint_states/joint_right_leg1_leg2/effort" custom_transform="noTransform" B="14" G="127" R="255"/>
    <transform value="noTransform"/>
   </plot>
  </plotmatrix>
  <currentPlotMatrix index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="200"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin status="idle" ID="ROS /rosout Visualization"/>
  <plugin status="idle" ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets>
  <snippet name="1st_derivative" language="JS">
   <global>var prevX = 0
var prevY = 0</global>
   <equation>dx = time - prevX
dy = value - prevY
prevX = time
prevY = value

return dy/dx</equation>
  </snippet>
  <snippet name="1st_order_lowpass" language="JS">
   <global>var prevY = 0
var alpha = 0.1</global>
   <equation>prevY = alpha * value + (1.-alpha) * prevY

return prevY</equation>
  </snippet>
  <snippet name="2D_velocity" language="JS">
   <global>var prev_x = 0
var prev_y = 0
var prev_t = 0</global>
   <equation>X = $$your_odometry/position/x$$
Y = $$your_odometry/position/y$$

var dist = sqrt( (X-prev_x)*(X-prev_x) + (Y-prev_y)*(Y-prev_y) )
var dT = time - prev_t

prev_x = X
prev_y = Y
prev_t = time

return dist / dT</equation>
  </snippet>
  <snippet name="average_two_curves" language="JS">
   <global></global>
   <equation>a = $$PLOT_A$$
b = $$PLOT_B$$

return (a+b)/2</equation>
  </snippet>
  <snippet name="integral" language="JS">
   <global>var integral = 0</global>
   <equation>integral += value
return integral</equation>
  </snippet>
  <snippet name="rad_to_deg" language="JS">
   <global></global>
   <equation>return value*180/3.1417</equation>
  </snippet>
  <snippet name="remove_offset" language="JS">
   <global>var is_first = true
var first_value = 0</global>
   <equation>if (is_first)
{
  is_first = false
  first_value = value
}

return value - first_value</equation>
  </snippet>
  <snippet name="yaw_from_quaternion" language="JS">
   <global>// source: https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles

function quaternionToYaw(x, y, z, w)
{
  // yaw (z-axis rotation)
  t1 = 2.0 * (w * z + x * y);
  t2 = 1.0 - 2.0 * (y * y + z * z);
  yaw = Math.atan2(t1, t2);

  return yaw
}</global>
   <equation>return quaternionToYaw(x, y, z, w);</equation>
  </snippet>
 </snippets>
 <!-- - - - - - - - - - - - - - - -->
</root>

